### ISSUE TITLE
Make Zend Lucene search "accent-insensitive"

### ISSUE THREAD

**Issue by pgorod** [salesagility/SuiteCRM#1133]
This is regarding the AOD global search, accessible from the top toolbar. It is powered by Zend Lucene Search package, if I am not mistaken.

For most non-english speakers, characters with accents (diacritics) are very common, and we are now used to neglect them when searching (Google is clever enough to handle the situation, and we are now used to that!). Accents are troublesome to insert with the keyboard, and the computer should be able to take care of them. So, the search should be insensitive to these characters (so "e" should match all of these: "eéèëê").

I searched online about this issue and found that the solution isn't available within the Zend package, but could work (I haven't tested) by changing the way the strings are passed on to the indexer.

Please see http://stackoverflow.com/questions/3346945/zend-search-lucene-and-accented-characters
for an explanation, and a useful code snippet.

I'm afraid I'm not familiar with how this is implemented in SuiteCRM but I was hoping this could be a simple change for someone who knows their way around this codebase. It would make a big difference for many users around the world.

**Comment by pgorod** [comment]
Just for practical purposes, here is the snippet from that site I linked above:
```
function normalize ($string){
    $a = 'ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝÞßàáâãäåæçèéêëìíîïðñòóôõöøùúûýýþÿŔŕ';
    $b = 'aaaaaaaceeeeiiiidnoooooouuuuybsaaaaaaaceeeeiiiidnoooooouuuyybyRr';
    $string = utf8_decode($string);
    $string = strtr($string, utf8_decode($a), $b);
    $string = strtolower($string);
    return utf8_encode($string);
}
$passToIndexer = normalize(" Aïcha ");
```
 I think the idea is to pass this changed string into the indexer in the first place; after that, the search would not mind accents anymore. If we're lucky, this could be a change only in one line in the code (apart from adding this function).

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Make Zend Lucene search "accent-insensitive"](https://trello.com/c/cOqyJr3u/111-make-zend-lucene-search-accent-insensitive)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
